sdv/single_table/base.py

```diff
@@ -204,6 +204,18 @@ def _validate(self, data):
         """
         return []
 
+    def _validate_constraints(self, data):
+        """Validate that the data satisfies the constraints."""
+        errors = []
+        constraints = self._data_processor._load_constraints()
+        for constraint in constraints:
+            try:
+                constraint.fit(data)
```

Comment:
So this means we fit constraints twice right? Since they get fit here too 
https://github.com/sdv-dev/SDV/blob/0e39f0f9490b704fe3e9bf48d2795cca8870c341/sdv/data_processing/data_processor.py#L289-L294

Is there a way we can avoid this?

Reply:
> It looks like we also don't ever validate the metadata against the constraints with this method
> 
> https://github.com/sdv-dev/SDV/blob/0e39f0f9490b704fe3e9bf48d2795cca8870c341/sdv/constraints/base.py#L156
> 
> 
> It doesn't seem like that has any references. We might need to add that too

It is being validated when we validate the `constraint dict` when adding a constraint: 
https://github.com/sdv-dev/SDV/blob/master/sdv/data_processing/data_processor.py#L252